### PR TITLE
Modules separation: rename module manageiq -> manageiq_provider

### PR DESCRIPTION
**Pull request by dkorn**
Needed changes in add_provider example playbook, setup and tests.
Also, updates for README.md.

### CODE REVIEW

**Comment by dkorn** [comment]
@jaryn, @cben, @enoodle, @kwoodson,@simon3z please review

File: README.md [hunk 1]
```diff
@@ -15,7 +15,9 @@ ManageIQ Python API Client package [manageiq-api-client-python] (https://github.
 
 ### Getting Started
 
-Currently, the module supports adding an OpenShift containers provider to manageiq.
+###### manageiq_provider module
+
+The `manageiq_provider.py` module currently supports adding, updating and deleting an OpenShift containers provider to manageiq.
```
Comment (kwoodson) [inline README.md]: Can remove the word 'containers'.

Comment (cben) [reply README.md]: I'd also s/manageiq_provider.py/manageiq_provider/, as user cares what the module is called in playbooks, not which file implements it.  </nitpick>

**Review by cben — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (dkorn) [reply README.md]: @kwoodson - we usually add it when formally describing OpenShift, but I don't mind removing it, thanks.
@cben - I agree, thanks

**Comment by dkorn** [comment]
fixed all comment, merging.

**Pull request merged**